### PR TITLE
docs: Improve wording on stating that a method does not work in-place

### LINKED ISSUE #346: docs: Improve wording on stating that a method does not work in-place
### Describe the bug

For methods that return a copy, rather than working in-place, I believe the standard wording in docstrings is:
```py
"""
Returns a new table that/where/with [...].

[...]
"""
```

Currently, we have this sentence in most of our docstrings:
```py
"""
[...]
This table is not modified.
[...]
"""
```
Might be unclear for some users what "this table" refers to.

**This rule should also be added to the guidelines.**

---

Certainly wrong is [this](https://github.com/Safe-DS/Stdlib/blob/main/src/safeds/data/tabular/containers/_table.py#L1126-L1130):
```py
    def tag_columns(self, target_name: str, feature_names: list[str] | None = None) -> TaggedTable:
        """
        Mark the columns of the table as target column or feature columns. The original table is not modified.

        This table is not modified.
        [...]
        """
```

### Expected behavior

Agree on a standard wording to use everywhere.

Definitely fix the redundancy in `tag_columns` in `src/safeds/data/tabular/containers/_table.py`.

### PR DESCRIPTION
**Pull request by zzril**
Closes #346

### Summary of Changes

Updated the docstrings of methods in `src/safeds/data` to state that copies are returned, with the original object left unmodified.

### CODE REVIEW

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 1]
```diff
@@ -290,9 +290,9 @@ def add_column(self, column: Column) -> TaggedTable:
 
     def add_columns(self, columns: list[Column] | Table) -> TaggedTable:
         """
-        Add multiple columns to the table, as neither target nor feature columns.
+        Add multiple columns to the table, as neither target nor feature columns, and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:293]:
```suggestion
        Return a new `TaggedTable` with multiple added columns, as neither target nor feature columns.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 2]
```diff
@@ -319,9 +319,9 @@ def add_columns(self, columns: list[Column] | Table) -> TaggedTable:
 
     def add_row(self, row: Row) -> TaggedTable:
         """
-        Add a row to the table.
+        Add a row to the table and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:322]:
```suggestion
        Return a new `TaggedTable` with an added Row attached.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 3]
```diff
@@ -342,9 +342,9 @@ def add_row(self, row: Row) -> TaggedTable:
 
     def add_rows(self, rows: list[Row] | Table) -> TaggedTable:
         """
-        Add multiple rows to the table.
+        Add multiple rows to the table and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:345]:
```suggestion
        Return a new `TaggedTable` with multiple added Rows attached.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 4]
```diff
@@ -423,9 +423,9 @@ def keep_only_columns(self, column_names: list[str]) -> TaggedTable:
 
     def remove_columns(self, column_names: list[str]) -> TaggedTable:
         """
-        Remove the given column(s) from the table.
+        Remove the given column(s) from the table and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:426]:
```suggestion
        Return a new `TaggedTable` with the given column(s) removed from the table.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 5]
```diff
@@ -461,9 +461,9 @@ def remove_columns(self, column_names: list[str]) -> TaggedTable:
 
     def remove_columns_with_missing_values(self) -> TaggedTable:
         """
-        Remove every column that misses values.
+        Remove every column that misses values and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:464]:
```suggestion
        Return a new `TaggedTable` with every column that misses values removed.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 6]
```diff
@@ -493,9 +493,9 @@ def remove_columns_with_missing_values(self) -> TaggedTable:
 
     def remove_columns_with_non_numerical_values(self) -> TaggedTable:
         """
-        Remove every column that contains non-numerical values.
+        Remove every column that contains non-numerical values and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:496]:
```suggestion
        Return a new `TaggedTable` with every column that contains non-numerical values removed.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 7]
```diff
@@ -525,9 +525,9 @@ def remove_columns_with_non_numerical_values(self) -> TaggedTable:
 
     def remove_duplicate_rows(self) -> TaggedTable:
         """
-        Remove all row duplicates.
+        Remove all row duplicates and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:528]:
```suggestion
        Return a new `TaggedTable` with all row duplicates removed
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 8]
```diff
@@ -559,13 +559,13 @@ def remove_rows_with_missing_values(self) -> TaggedTable:
 
     def remove_rows_with_outliers(self) -> TaggedTable:
         """
-        Remove all rows from the table that contain at least one outlier.
+        Remove all rows from the table that contain at least one outlier and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:562]:
```suggestion
        Return a new `TaggedTable` with all rows that contain at least one outlier removed.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 9]
```diff
@@ -580,9 +580,9 @@ def remove_rows_with_outliers(self) -> TaggedTable:
 
     def rename_column(self, old_name: str, new_name: str) -> TaggedTable:
         """
-        Rename a single column.
+        Rename a single column and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:583]:
```suggestion
        Return a new `TaggedTable` with a single column renamed.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 10]
```diff
@@ -617,13 +617,13 @@ def rename_column(self, old_name: str, new_name: str) -> TaggedTable:
 
     def replace_column(self, old_column_name: str, new_columns: list[Column]) -> TaggedTable:
         """
-        Replace the specified old column by a list of new columns.
+        Replace the specified old column by a list of new columns and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:620]:
```suggestion
        Return a new `TaggedTable` with the specified old column replaced by a list of new columns.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 11]
```diff
@@ -674,9 +674,9 @@ def replace_column(self, old_column_name: str, new_columns: list[Column]) -> Tag
 
     def shuffle_rows(self) -> TaggedTable:
         """
-        Shuffle the table randomly.
+        Shuffle the table randomly and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:677]:
```suggestion
        Return a new `TaggedTable` with randomly shuffled rows of this table.
```

File: src/safeds/data/tabular/containers/_tagged_table.py [hunk 12]
```diff
@@ -795,9 +795,9 @@ def sort_rows(self, comparator: Callable[[Row, Row], int]) -> TaggedTable:
 
     def transform_column(self, name: str, transformer: Callable[[Row], Any]) -> TaggedTable:
         """
-        Transform provided column by calling provided transformer.
+        Transform the provided column by calling the provided transformer and return a new `TaggedTable`.
```
Comment (Marsmaennchen221) [inline src/safeds/data/tabular/containers/_tagged_table.py:798]:
```suggestion
        Return a new `TaggedTable` with the provided column transformed by calling the provided transformer.
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]
If you write what a method does first and than write "return a new <object>" it sound like you could be doing this operation on the original object and just copy afterwards. Although you specify below that the original object will not be modified I think it would better to clarify this directly.

File: src/safeds/data/tabular/containers/_tagged_table.py (hunk 7 above)
Comment (zzril) [reply src/safeds/data/tabular/containers/_tagged_table.py:528]: The dot should stay, though.

**Comment by zzril** [comment]
> If you write what a method does first and than write "return a new " it sound like you could be doing this operation on the original object and just copy afterwards. Although you specify below that the original object will not be modified I think it would better to clarify this directly.

Done.

File: src/safeds/data/image/containers/_image.py [hunk 13]
```diff
@@ -347,19 +361,21 @@ def add_gaussian_noise(self, standard_deviation: float) -> Image:
...
-        factor: float
-            If factor > 1, increase contrast of image.
-            If factor = 1, no changes will be made.
-            If factor < 1, make image greyer.
-            Has to be bigger than or equal to 0 (gray).
+         factor: float
+             If factor > 1, increase contrast of image.
+             If factor = 1, no changes will be made.
+             If factor < 1, make image greyer.
+             Has to be bigger than or equal to 0 (gray).
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:374]:
```suggestion
        factor: float
            If factor > 1, increase contrast of image.
            If factor = 1, no changes will be made.
            If factor < 1, make image greyer.
            Has to be bigger than or equal to 0 (gray).
```

File: src/safeds/data/image/containers/_image.py [hunk 14]
```diff
@@ -347,19 +361,21 @@ def add_gaussian_noise(self, standard_deviation: float) -> Image:
 
     def adjust_contrast(self, factor: float) -> Image:
         """
-        Adjust Contrast of image.
+        Return a new `Image` with adjusted contrast.
+
+         The original image is not modified.
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:366]:
```suggestion
        The original image is not modified.
```

File: src/safeds/data/image/containers/_image.py [hunk 15]
```diff
@@ -347,19 +361,21 @@ def add_gaussian_noise(self, standard_deviation: float) -> Image:
...
+         factor: float
+             If factor > 1, increase contrast of image.
+             If factor = 1, no changes will be made.
+             If factor < 1, make image greyer.
+             Has to be bigger than or equal to 0 (gray).
 
         Returns
         -------
-        New image with adjusted contrast.
+         New image with adjusted contrast.
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:378]:
```suggestion
        image: Image
            New image with adjusted contrast.
```

File: src/safeds/data/image/containers/_image.py [hunk 16]
```diff
@@ -220,20 +222,24 @@ def resize(self, new_width: int, new_height: int) -> Image:
...
+        Return a new `Image` that is converted to grayscale.
+
+         The original image is not modified.
 
         Returns
         -------
-        grayscale_image : Image
-            The grayscale image.
+         grayscale_image : Image
+             The grayscale image.
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:232]:
```suggestion
        grayscale_image : Image
            The grayscale image.
```

File: src/safeds/data/image/containers/_image.py [hunk 17]
```diff
@@ -220,20 +222,24 @@ def resize(self, new_width: int, new_height: int) -> Image:
 
     def convert_to_grayscale(self) -> Image:
         """
-        Convert the image to grayscale.
+        Return a new `Image` that is converted to grayscale.
+
+         The original image is not modified.
```
Comment (Marsmaennchen221) [inline src/safeds/data/image/containers/_image.py:227]:
```suggestion
        The original image is not modified.
```

**Review by Marsmaennchen221 — APPROVED** [review]

**Pull request merged**